sdk/go/common/workspace/plugins_test.go

```diff
@@ -1080,15 +1146,74 @@ func TestBundledPluginSearch(t *testing.T) {
 	err = os.WriteFile(ambientPath, []byte{}, 0o700) //nolint: gosec
 	require.NoError(t, err)
 
+	d := &mockSink{t: t}
+
 	// Lookup the plugin with ambient search turned on
 	t.Setenv("PULUMI_IGNORE_AMBIENT_PLUGINS", "false")
-	path, err := GetPluginPath(LanguagePlugin, "mock", nil, nil)
+	path, err := GetPluginPath(d, LanguagePlugin, "mock", nil, nil)
 	require.NoError(t, err)
 	assert.Equal(t, ambientPath, path)
 
 	// Lookup the plugin with ambient search turned off
 	t.Setenv("PULUMI_IGNORE_AMBIENT_PLUGINS", "true")
-	path, err = GetPluginPath(LanguagePlugin, "mock", nil, nil)
+	path, err = GetPluginPath(d, LanguagePlugin, "mock", nil, nil)
+	require.NoError(t, err)
+	assert.Equal(t, bundledPath, path)
+}
+
+//nolint:paralleltest // modifies environment variables
+func TestAmbientPluginsWarn(t *testing.T) {
+	// Create a fake plugin in the path
+	pathDir := t.TempDir()
+	t.Setenv("PATH", pathDir)
+	ambientPath := filepath.Join(pathDir, "pulumi-resource-mock")
+	err := os.WriteFile(ambientPath, []byte{}, 0o700) //nolint: gosec
+	require.NoError(t, err)
```

Comment:
This is fine as-is, but you may find it easier to not have a mock sink and instead use a `diag.DefaultSink` with a buffer for stderr.

```go
var stderr bytes.Buffer
d := diag.DefaultSink(
	iotest.LogWriter(t), // stdout
	&stderr,
	diag.FormatOptions{Color: "never"},
)

// ...

assert.Regexp(t, `using \S+/pulumi-resource-mock from \$PATH` stderr.String())
```

This way you're also validating the final message and where it's printed instead of the printf string template.

([iotest.LogWriter comes from here](https://github.com/pulumi/pulumi/blob/d96533d46a238cc506bb13b6e560637a05e40577/sdk/go/common/testing/iotest/log_writer.go#L56).)